### DISCUSSION TITLE
Supabase with Prisma workflow

### DISCUSSION THREAD

**Discussion by NanoCode012** [prisma/prisma#14292] [Q&A]
Hello everyone, I am loving Prisma the more I learn about it, especially the `schema.prisma` which allows me to easily modify table structures and see relationships.

However, I've met a lot of bottlenecks along the way and found solutions for them. I want to ask if others have managed to get a working project up and running with Supabase and would like to share their experience?

I have outlined all my problems (with references and workarounds) in the Supabase discussions https://github.com/supabase/supabase/discussions/7659 .

The main problems and solutions are:
- (Point 1&2) Cross schema issue. Required to create a separate table in `public.user` with a trigger that updates based on `auth.users`. The trigger also has to be created externally and applied via `psql`
- (Point 3) Needing to create dummy function `public.auth_id()` which is replaced by an external `psql`
- (Point 4) Incremental ID is not working
- (Point 5) Grant issues with prisma-created tables
- (Point 7) Needing to inject `current_user_id` into middleware

Are there things that I have missed? Am I approaching this the right way? It would be great if we could consolidate all these issues across discussions into one place.

**Comment by ruheni** [comment]
Hey 👋🏽

Thanks for raising this. I have reached out to our Product and Engineering team for further clarification.

> (Points 1&2) Cross schema issue

We're currently looking into supporting querying multiple database schemas, and we would appreciate your feedback on this [GitHub issue](https://github.com/prisma/prisma/issues/1122).

> (Point 5) Grant issues with prisma-created tables

I recommend storing the grants in your migration history to ensure you don't lose track of it.

**Reply by NanoCode012** [reply]
Hey, @ruheni ! Thanks for your comment! I am currently following that issue as well and am excited that this is a WIP for implementation (https://github.com/prisma/prisma/issues/1122#issuecomment-1152193257).

> I recommend storing the grants in your migration history to ensure you don't lose track of it.

For grants, I have not reached that stage in my own testing (other issues cropped up first leading to my posts), however, I have noted others facing the issue, so I also put it on the list for point of reference.

For now, I am trying to look for and understand other's experience combining the two technology together. For ex, how they deal with the cross-schema issue, do they create a dummy function as said above and replace it manually via `psql` or are there other alternatives?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
